google/cloud/bigtable/cell.h

```diff
@@ -60,6 +60,28 @@ class Cell {
         value_(google::cloud::bigtable::internal::AsBigEndian64(value)),
         labels_(std::move(labels)) {}
 
+  /// Create a cell and fill it with data, but with empty labels.
+  Cell(std::string row_key, std::string family_name,
+       std::string column_qualifier, std::int64_t timestamp, std::string value)
+      : row_key_(std::move(row_key)),
+        family_name_(std::move(family_name)),
+        column_qualifier_(std::move(column_qualifier)),
+        timestamp_(timestamp),
+        value_(std::move(value)),
+        labels_{} {}
+
+  /// Create a Cell and fill it with bigendian 64 bit value, but with empty
+  /// labels.
+  Cell(std::string row_key, std::string family_name,
+       std::string column_qualifier, std::int64_t timestamp,
+       bigendian64_t value)
+      : row_key_(std::move(row_key)),
```

Comment:
Rather than setting the data members in each constructor's initialization list, I think we should use forwarding constructors. In other words, this class has 6 fields, and the first constructor listed exactly sets all those fields. The other 3 constructors are just slight variations on that first constructor (in some other languages, the first constructor is called the "designated initializer" or "designated constructor"). So, each of the remaining three constructors should just call the first constructor adding a reasonable default for any unset fields. So I'm thinking something like this:

```cc
 52   /// Create a Cell and fill it with bigendian 64 bit value.
 53   Cell(std::string row_key, std::string family_name,
 54        std::string column_qualifier, std::int64_t timestamp,
 55        bigendian64_t value, std::vector<std::string> labels)
 56       : Cell(std::move(row_key), std::move(family_name),
 57              std::move(column_qualifier), timestamp,
 58              internal::AsBigEndian64(value), std::move(labels)) {}
 59 
 60   /// Create a cell and fill it with data, but with empty labels.
 61   Cell(std::string row_key, std::string family_name,
 62        std::string column_qualifier, std::int64_t timestamp, std::string value)
 63       : Cell(std::move(row_key), std::move(family_name),
 64              std::move(column_qualifier), timestamp, std::move(value),
 65              std::move(labels)) {}
 66 
 67   /// Create a Cell and fill it with bigendian 64 bit value, but with empty
 68   /// labels.
 69   Cell(std::string row_key, std::string family_name,
 70        std::string column_qualifier, std::int64_t timestamp,
 71        bigendian64_t value)
 72       : Cell(std::move(row_key), std::move(family_name),
 73              std::move(column_qualifier), timestamp, std::move(value), {}) {}

```

Reply:
Thanks for the suggestion. Fixed. However, I believe it's actually called a delegating constructor.

Reply:
Yup, you're right. thanks. I knew "forwarding" didn't sound right when I was typing it.